api/src/main/kotlin/se/ansman/kotshi/KotshiUtils.kt

```diff
@@ -39,4 +44,44 @@ object KotshiUtils {
             }
         } as T
     }
+
+    @JvmStatic
+    fun JsonReader.nextFloat(): Float {
+        val value = nextDouble().toFloat()
+        // Double check for infinity after float conversion; many doubles > Float.MAX
+        if (!isLenient && value.isInfinite()) {
+            throw JsonDataException("JSON forbids NaN and infinities: $value at path $path")
+        }
+        return value
+    }
+
+    @JvmStatic
+    fun JsonReader.nextByte(): Byte = nextIntInRange("a byte", -128, 265).toByte()
```

Comment:
On the other hand, it would mean you couldn't read e.g a JSON value of `255` as a byte. Being that the JVM only has signed types, this might be OK. Perhaps there should be a qualifier for an "unsigned" byte (parsed as an int in 0..255, but represented as a byte in the JVM, -128..127).

Reply:
This implementation is taken from Moshi itself, we need it to be the same

Reply:
Well, OK, but Moshi uses 255 as the max at least.

Reply:
Oops, that's a bug